src/app/DownloadDelegate.qml

```diff
@@ -132,71 +167,59 @@ ListItem {
-                opacity: paused ? 0.5 : 1
-            }
-        }
 
-        Column {
-            SlotsLayout.position: SlotsLayout.Trailing
-            spacing: units.gu(1)
-            width: (incomplete && !error.visible) ? cancelButton.width : 0
+                IndeterminateProgressBar {
+                    id: progressBar
 
-            Button {
-                id: cancelButton
-                visible: incomplete && !error.visible
-                text: i18n.tr("Cancel")
-                onClicked: {
-                    if (download) {
-                        download.cancel()
-                        cancelled()
-                    }
+                    Layout.fillWidth: true
+                    implicitHeight: units.gu(0.5)
+                    progress: downloadDelegate.progress
+                    // Work around UDM bug #1450144
+                    indeterminateProgress: progress < 0 || progress > 100
+                    opacity: paused ? 0.5 : 1
                 }
-            }
-
-            Label {
-                visible: !progressBar.indeterminateProgress && incomplete && !error.visible
-                width: cancelButton.width
-                horizontalAlignment: Text.AlignHCenter
-                textSize: Label.Small
-                // TRANSLATORS: %1 is the percentage of the download completed so far
-                text: i18n.tr("%1%").arg(progressBar.progress)
-                opacity: paused ? 0.5 : 1
-            }
-
-            Button {
-                visible: incomplete && ! paused && ! error.visible
-                text: i18n.tr("Pause")
-                width: cancelButton.width
-                onClicked: {
-                    if (download) {
-                        download.pause()
+                
+                RowLayout {
+                    Layout.fillWidth: true
+                    implicitHeight: units.gu(4)
+                    
+                    Label {
+                        horizontalAlignment: Text.AlignHCenter
+                        textSize: Label.Small
+                        text: download ? FileUtils.formatBytes(download.receivedBytes) + " / " + FileUtils.formatBytes(download.totalBytes) : i18n.tr("Unknown")
+                        opacity: paused ? 0.5 : 1
                     }
-                }
-            }
-
-            Button {
-                visible: incomplete && paused && ! error.visible
-                text: i18n.tr("Resume")
-                width: cancelButton.width
-                onClicked: {
-                    if (download) {
-                        download.resume()
+                    
+                    Label {
+                        horizontalAlignment: Text.AlignHCenter
+                        textSize: Label.Small
+                        // TRANSLATORS: %1 is the percentage of the download completed so far
+                        text: "(" + i18n.tr("%1/s").arg(FileUtils.formatBytes(downloadDelegate.speed)) + ")"
```

Comment:
It doesn't seem like there is any translatable text here. Why is it a translatable string?

Reply:
Hmm, the comment is a copy/paste mistake 😅
Can't remember what my reason behind or if this wasn't intended but I think it's because other languages may also display `/s` differently?